File: Maths_And_Stats/Perfect_Number/PerfectNumber.py


numlist = []
num1 = int(input("number: \n"))

for i in range(1, num1 - 1):
    y = num1%i
    if(y == 0):
        numlist.append(i)
x = 0
for i in range(len(numlist)):
    x = x + numlist[i]

if(x == num1):
    print(num1, " is a perfect number!")

else:
    print(num1, " is not a perfect number")

File: Maths/Number-Theory/Perfect_Number/perfectnumber.php
function isPerfect($n) { 
    $sum = 1; 
    for ($i = 2; $i * $i <= $n; $i++) { 
        if ($n % $i == 0) { 
            if($i * $i != $n) {
                $sum = $sum + $i + (int)($n / $i); 
            } else {
                $sum = $sum + $i;
            } 
        }
    }  
    if ($sum == $n && $n != 1) {
        echo "$n is a perfect number"; 
    } else {
        echo "$n is not a perfect number"; 
    }
}
isPerfect(6);